pkg/cache/v3/delta.go

```diff
@@ -59,19 +73,13 @@ func createDeltaResponse(ctx context.Context, req *DeltaRequest, state stream.St
 			} else {
 				// We track non-existent resources for non-wildcard streams until the client explicitly unsubscribes from them.
 				nextVersionMap[name] = ""
+				if prevVersion != "" {
+					toRemove = append(toRemove, name)
+				}
 			}
 		}
 	}
 
-	// Compute resources for removal regardless of the request type
-	for name, prevVersion := range state.GetResourceVersions() {
-		// The prevVersion != "" check is in place to make sure we are only sending an update to the client once right after it is removed.
-		// If the client decides to keep the subscription we skip the add for every subsequent response.
-		if _, ok := resources.resourceMap[name]; !ok && prevVersion != "" {
-			toRemove = append(toRemove, name)
-		}
-	}
-
```

Comment:
Why did you delete this part?

Reply:
I merged this part with the iterations above. In the case of non-wildcard, this avoids an entire map iteration that is already done (line 77). For wildcard code is just moved (line 56)